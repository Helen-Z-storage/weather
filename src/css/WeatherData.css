
.data_space {
    width: 100%; /* Full-width */
}

.curr_weather{
  width: 7em;
  margin: 10px;
  border-radius: 50%;
  border: 1px solid #DCDCDC;
}

.city_name {
    font-size: 2em;
    font-weight: bolder;
    float: left;
    padding: 0.3em 0em;
  }
  
.country_name {
  font-size: 1em;
  float: left;
  padding: 0.3em 0em;
}

.city_data {
    padding-top: 5px;
  }
  