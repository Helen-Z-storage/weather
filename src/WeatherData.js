import React from 'react';
import './WeatherData.css';

function WeatherData(props) {
    const {city, name, avg} = props;

    return (
        <tr>
            <td>
                <img alt={`${name} profile`} src={city.pic} />
            </td>
            <td>
                <ul className='data_space'>
                    <li key="0"><div className='stu_name' style={{width:"100%"}}>{name}</div></li>
                    <li key="1" className='stu_data'>{`Email: ${city.email}`}</li>
                    <li key="2" className='stu_data'>{`Company: ${city.company}`}</li>
                    <li key="3" className='stu_data'>{`Skill: ${city.skill}`}</li>
                    <li key="4" className='stu_data'>{`Average: ${Math.round(avg * 1000) / 1000}%`}</li>
                </ul>
            </td>
        </tr>
    );
}

export default WeatherData;
