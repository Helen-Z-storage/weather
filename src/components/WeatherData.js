import React from 'react';
import '../css/WeatherData.css';

import { connect } from "react-redux";
import * as uiActions from "./redux/actions/uiActions";
import * as pageActions from "./redux/actions/pageActions";

function WeatherData(props) {
    const {city, city_name, avg} = props;

    return (
        <tr>
            <td>
                <img alt={`${city_name} profile`} src={city.pic} />
            </td>
            <td>
                <ul className='data_space'>
                    <li key="0"><div className='city_name' style={{width:"100%"}}>{city_name}</div></li>
                    <li key="1" className='city_data'>{`Email: ${city.email}`}</li>
                    <li key="2" className='city_data'>{`Company: ${city.company}`}</li>
                    <li key="3" className='city_data'>{`Skill: ${city.skill}`}</li>
                    <li key="4" className='city_data'>{`Average: ${Math.round(avg * 1000) / 1000}%`}</li>
                </ul>
            </td>
        </tr>
    );
}

export default connect (
    (state) => {
      return {
        ui: state.ui,
        posts: state.posts
      }
    })(WeatherData);
