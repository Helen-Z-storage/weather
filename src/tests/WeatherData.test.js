import React from 'react'
import { create } from "react-test-renderer";

import '@testing-library/jest-dom/extend-expect' // expect function

import WeatherData from '../components/WeatherData';

// required prop data of create component
const currCity =  {
    country: "Abkhazia",
    city: "Sukhumi",
    lat: 43.0033629,
    lon: 41.0192741,
    tags: ",",
    icon: "https://openweathermap.org/img/wn/03d@2x.png",
    curr_forecast: {
        timestamp: "2023-01-18 09:00:00",
        temp: 11.57,
        feels_like: 10.06,
        temp_min: 11.57,
        temp_max: 14.54,
        icon: "https://openweathermap.org/img/wn/03d@2x.png"
    },
    detail_forecast: [
        {
            timestamp: "2023-01-18 09:00:00",
            temp: 11.57,
            feels_like: 10.06,
            temp_min: 11.57,
            temp_max: 14.54,
            icon: "https://openweathermap.org/img/wn/03d@2x.png"
        },
        {
            timestamp: "2023-01-18 12:00:00",
            temp: 13.2,
            feels_like: 11.83,
            temp_min: 13.2,
            temp_max: 14.75,
            icon: "https://openweathermap.org/img/wn/03d@2x.png"
        },
        {
            timestamp: "2023-01-18 15:00:00",
            temp: 12.26,
            feels_like: 10.95,
            temp_min: 12.26,
            temp_max: 12.26,
            icon: "https://openweathermap.org/img/wn/04n@2x.png"
        },
        {
            timestamp: "2023-01-18 18:00:00",
            temp: 11.37,
            feels_like: 9.97,
            temp_min: 11.37,
            temp_max: 11.37,
            icon: "https://openweathermap.org/img/wn/04n@2x.png"
        },
        {
            timestamp: "2023-01-18 21:00:00",
            temp: 11.01,
            feels_like: 9.55,
            temp_min: 11.01,
            temp_max: 11.01,
            icon: "https://openweathermap.org/img/wn/04n@2x.png"
        },
        {
            timestamp: "2023-01-19 00:00:00",
            temp: 10.28,
            feels_like: 8.7,
            temp_min: 10.28,
            temp_max: 10.28,
            icon: "https://openweathermap.org/img/wn/04n@2x.png"
        },
        {
            timestamp: "2023-01-19 03:00:00",
            temp: 9.95,
            feels_like: 8.9,
            temp_min: 9.95,
            temp_max: 9.95,
            icon: "https://openweathermap.org/img/wn/02n@2x.png"
        },
        {
            timestamp: "2023-01-19 06:00:00",
            temp: 11.08,
            feels_like: 9.47,
            temp_min: 11.08,
            temp_max: 11.08,
            icon: "https://openweathermap.org/img/wn/02d@2x.png"
        }
    ]
};

describe('testing WeatherData component', () => {
    // snampshot testing, for non-state component only
    test("Matches the snapshot", () => {
        // create component
        const cityData = create(
            <WeatherData currCity={currCity}/>
        );

        // expecting output
        expect(cityData.toJSON()).toMatchSnapshot();
    });
});