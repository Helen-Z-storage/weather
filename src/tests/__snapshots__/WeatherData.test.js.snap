// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing WeatherData component Matches the snapshot 1`] = `
<tr>
  <td>
    <img
      alt="Sukhumi weather icon"
      className="curr_weather"
      src="https://openweathermap.org/img/wn/03d@2x.png"
    />
  </td>
  <td>
    <ul
      className="data_space"
    >
      <li>
        <div
          className="city_name"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          Sukhumi
        </div>
      </li>
      <li>
        <div
          className="country_name"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          Abkhazia
        </div>
      </li>
    </ul>
  </td>
</tr>
`;
